weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTableButtons.tsx

```diff
@@ -0,0 +1,315 @@
+
+type SelectionState = {
+  allChecked?: boolean;
+  selectedChecked?: boolean;
+  limitChecked?: boolean;
+  limit?: string | null;
+};
+
+export const ExportSelector = ({
+  selectedCalls,
+  numTotalCalls,
+  visibleColumns,
+  disabled,
+  callQueryParams,
+  rightmostButton = false,
+}: {
+  selectedCalls: string[];
+  numTotalCalls: number;
+  visibleColumns: string[];
+  callQueryParams: {
+    entity: string;
+    project: string;
+    filter: WFHighLevelCallFilter;
+    gridFilter: GridFilterModel;
+    gridSort?: GridSortModel;
+  };
+  disabled: boolean;
+  rightmostButton?: boolean;
+}) => {
+  const [selectionState, setSelectionState] = useState<SelectionState>({
+    allChecked: true,
+  });
+
+  // Popover management
+  const ref = useRef<HTMLDivElement>(null);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const onClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(anchorEl ? null : ref.current);
+  };
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popper' : undefined;
+
+  // Call download query
+  const {useCallsExport} = useWFHooks();
+  const download = useCallsExport();
+  const {sortBy, lowLevelFilter, filterBy} = useFilterSortby(
+    callQueryParams.filter,
+    callQueryParams.gridFilter,
+    callQueryParams.gridSort
+  );
+
+  const onClickDownload = (contentType: ContentType) => {
+    lowLevelFilter.callIds = selectionState.selectedChecked
+      ? selectedCalls
+      : undefined;
+    // TODO(gst): allow user to specify offset?
+    const offset = 0;
+    const limit = selectionState.limitChecked
+      ? Math.min(MAX_EXPORT, parseInt(selectionState.limit ?? '100', 10))
+      : MAX_EXPORT;
+    download(
+      callQueryParams.entity,
+      callQueryParams.project,
+      contentType,
+      lowLevelFilter,
+      limit,
+      offset,
+      sortBy,
+      filterBy,
+      visibleColumns
+    ).then(blob => {
+      const fileExtension = fileExtensions[contentType];
+      const date = new Date().toISOString().split('T')[0];
+      const fileName = `weave_export_${callQueryParams.project}_${date}.${fileExtension}`;
+      initiateDownloadFromBlob(blob, fileName);
+      setAnchorEl(null);
+    });
+    setSelectionState({allChecked: true});
+  };
+
+  return (
+    <>
+      <span ref={ref}>
+        <Button
+          className={rightmostButton ? 'mr-16' : 'mr-4'}
+          icon="export-share-upload"
+          variant="ghost"
+          onClick={onClick}
+          disabled={disabled}
+        />
+      </span>
+      <Popover
+        id={id}
+        open={open}
+        anchorEl={anchorEl}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'center',
+        }}
+        onClose={() => {
+          setAnchorEl(null);
+          setSelectionState({allChecked: true});
+        }}>
+        <Tailwind>
+          <div className="min-w-[460px] p-12">
+            <div className="flex items-center pb-8">
```

Comment:
It would be nice to make this popup draggable as we have with others like the column visibility popup and should be pretty easy, see `ManageColumnsButton` for how it uses some helpers.